src/common/utils/registry/registry.go

```diff
@@ -38,28 +39,37 @@ func init() {
 	defaultHTTPTransport = &http.Transport{}
 
 	secureHTTPTransport = &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
+		// Proxy: http.ProxyFromEnvironment,
 		TLSClientConfig: &tls.Config{
 			InsecureSkipVerify: false,
 		},
 	}
 	insecureHTTPTransport = &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
+		// Proxy: http.ProxyFromEnvironment,
 		TLSClientConfig: &tls.Config{
 			InsecureSkipVerify: true,
 		},
 	}
 }
 
-// GetHTTPTransport returns HttpTransport based on insecure configuration
-func GetHTTPTransport(insecure ...bool) *http.Transport {
-	if len(insecure) == 0 {
+// GetHTTPTransport returns HttpTransport based on insecure, proxy configurations
+func GetHTTPTransport(configs ...bool) (t *http.Transport) {
+	log.Infof("[GetHTTPTransport.configs] %#v\n", configs)
+	if len(configs) == 0 {
 		return defaultHTTPTransport
 	}
-	if insecure[0] {
-		return insecureHTTPTransport
+	if configs[0] {
+		t = insecureHTTPTransport
+	} else {
+		t = secureHTTPTransport
 	}
-	return secureHTTPTransport
+	if len(configs) == 2 {
+		if configs[0] {
+			t.Proxy = http.ProxyFromEnvironment
```

Comment:
This cannot work as expected as the `t` here is a pointer, setting proxy for it will impact others